extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -185,20 +186,28 @@ public static void write(
                     final String parquetColumnName =
                             writeInstructions.getParquetColumnNameFromColumnNameOrDefault(groupingColumns[gci]);
                     final String groupingPath = groupingPathFactory.apply(parquetColumnName);
-                    cleanupPaths.add(groupingPath);
+                    groupingPaths.add(groupingPath);
                     tableInfoBuilder.addGroupingColumns(GroupingColumnInfo.of(parquetColumnName,
                             destDirPath.relativize(Paths.get(groupingPath)).toString()));
                     write(auxiliaryTables[gci], auxiliaryTables[gci].getDefinition(), writeInstructions, groupingPath,
                             Collections.emptyMap());
                 }
             }
             write(t, definition, writeInstructions, destPathName, incomingMeta, tableInfoBuilder);
+
+            // Invalidate old file handles so that we don't read from files which have been updated
+            TrackedFileHandleFactoryWithLookup.getInstance().invalidateHandles(new File(destPathName));
```

Comment:
This should probably happen up in `ParquetTools`, based on the other branch. We should invalidate all dest files *before* installing shadow files.